mod-source-record-storage-server/src/main/java/org/folio/services/handlers/actions/AbstractDeleteEventHandler.java

```diff
@@ -67,11 +68,18 @@ public CompletableFuture<DataImportEventPayload> handle(DataImportEventPayload p
   private void handlePayload(DataImportEventPayload payload, CompletableFuture<DataImportEventPayload> future) {
     var record = Json.decodeValue(payload.getContext().get(getRecordKey()), Record.class);
     LOG.info("handlePayload:: Handling 'delete' event for the record id = {}", record.getId());
-    recordService.updateRecordsState(record.getMatchedId(), RecordState.DELETED, payload.getTenant())
+    ParsedRecordDaoUtil.updateLeaderStatus(record.getParsedRecord(), 'd');
+    record.setState(Record.State.DELETED);
+    recordService.getRecordById(record.getId(), payload.getTenant())
+        .compose(optionalRecord -> optionalRecord
+          .map(existingRecord -> recordService.saveRecord(record, payload.getTenant()))
+          .orElse(Future.succeededFuture(null)))
```

Comment:
`updateRecordsState` method differs in behavior with `saveRecord` method.
`updateRecordsState` method set status to all generations of the records, instead of `saveRecord` that will create a new generation of a record
![image](https://github.com/folio-org/mod-source-record-storage/assets/5904625/dd63f438-f497-4fae-a026-566ab6bc08f6)